`over()` with descending `order_by` ignores `nulls_last` inside `group_by().agg()`

### Checks

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl

df = pl.DataFrame({
      "group": ["a", "a", "a"],
      "value": [1, 2, 3],
      "date":  [1, None, 2],
  })

expected = df.with_columns(
      pl.col("value").cum_sum().over("group", order_by="date", descending=True).alias("cs")
  )["cs"].to_list()

actual = df.group_by("group", maintain_order=True).agg(
      pl.col("value").cum_sum().over("group", order_by="date", descending=True)
  )["value"][0].to_list()

print(f"Expected: {expected}")  
print(f"Actual:   {actual}")   
assert expected == actual, "Bug: over() with descending=True misplaces nulls in agg() context"
```


### Log output

```shell
async thread count: 32
blocking thread count: 512
keys/aggregates are not partitionable: running default HASH AGGREGATION
Expected: [6, 2, 5]
Actual:   [5, 6, 2]
Traceback (most recent call last):
  File "/home/gauta/polars/repro_over_descending_nulls.py", line 21, in <module>
    assert expected == actual, "Bug: over() with descending=True misplaces nulls in agg() context"
           ^^^^^^^^^^^^^^^^^^
AssertionError: Bug: over() with descending=True misplaces nulls in agg() context
```

### Issue description

When `over()` with `order_by` is used inside `group_by().agg()`, results can be incorrect if `descending=True` and the `order_by` column contains nulls. This occurs with the default `nulls_last=False`, so users can hit it with a normal descending window expression without explicitly configuring null placement.

This is affecting non-scalar window expressions that use the groups-to-rows mapping path, such as `cum_sum`, `rank`, `shift`, and `diff`. Scalar aggregations such as `first`, `sum`, `mean`, `min`, and `max` appear unaffected because they use a different path.

I can reproduce this in eager and lazy execution and the same path also appears reachable in streaming.

### Expected behavior

`over()` with `order_by` inside `group_by().agg()` should produce the same result as `with_columns()`

### Installed versions

<details>

```
--------Version info---------
Polars:              1.40.1
Index type:          UInt32
Platform:            Linux-6.6.87.2-microsoft-standard-WSL2-x86_64-with-glibc2.39
Python:              3.13.11 | packaged by Anaconda, Inc. | (main, Dec 10 2025, 21:28:48) [GCC 14.3.0]
Runtime:             rt32

----Optional dependencies----
Azure CLI            <not installed>
adbc_driver_manager  <not installed>
altair               <not installed>
azure.identity       <not installed>
boto3                <not installed>
cloudpickle          <not installed>
connectorx           <not installed>
deltalake            <not installed>
fastexcel            <not installed>
fsspec               <not installed>
gevent               <not installed>
google.auth          <not installed>
great_tables         <not installed>
matplotlib           <not installed>
numpy                <not installed>
openpyxl             <not installed>
pandas               <not installed>
polars_cloud         <not installed>
pyarrow              <not installed>
pydantic             <not installed>
pyiceberg            <not installed>
sqlalchemy           <not installed>
torch                <not installed>
xlsx2csv             <not installed>
xlsxwriter           <not installed>

```

</details>

--- a/crates/polars-expr/src/expressions/window.rs
+++ b/crates/polars-expr/src/expressions/window.rs
@@ -1,4 +1,3 @@
-use std::cmp::Ordering;
 use std::fmt::Write;
 
 use arrow::array::PrimitiveArray;
@@ -845,14 +844,12 @@ impl PhysicalExpr for WindowExpr {
                                     let is_valid_b =
                                         unsafe { validity.get_bit_unchecked(in_group_idx_b) };
 
-                                    if !is_valid_a | !is_valid_b {
-                                        if !is_valid_a & !is_valid_b {
-                                            return Ordering::Equal;
+                                    if !(is_valid_a & is_valid_b) {
+                                        let mut cmp = is_valid_a.cmp(&is_valid_b);
+                                        if options.nulls_last {
+                                            cmp = cmp.reverse();
                                         }
-                                        return match (options.nulls_last, !is_valid_a) {
-                                            (false, true) | (true, false) => Ordering::Less,
-                                            _ => Ordering::Greater,
-                                        };
+                                        return cmp;
                                     }
 
                                     let order_a = unsafe { arr.get_unchecked(in_group_idx_a) };

--- a/crates/polars-expr/src/expressions/window.rs
+++ b/crates/polars-expr/src/expressions/window.rs
@@ -844,23 +844,22 @@ impl PhysicalExpr for WindowExpr {
                                         unsafe { validity.get_bit_unchecked(in_group_idx_a) };
                                     let is_valid_b =
                                         unsafe { validity.get_bit_unchecked(in_group_idx_b) };
+
+                                    if !is_valid_a | !is_valid_b {
+                                        if !is_valid_a & !is_valid_b {
+                                            return Ordering::Equal;
+                                        }
+                                        return match (options.nulls_last, !is_valid_a) {
+                                            (false, true) | (true, false) => Ordering::Less,
+                                            _ => Ordering::Greater,
+                                        };
+                                    }
+
                                     let order_a = unsafe { arr.get_unchecked(in_group_idx_a) };
                                     let order_b = unsafe { arr.get_unchecked(in_group_idx_b) };
 
-                                    if !is_valid_a & !is_valid_b {
-                                        return Ordering::Equal;
-                                    }
-
                                     let mut cmp = order_a.cmp(&order_b);
-                                    if !is_valid_a {
-                                        cmp = Ordering::Less;
-                                    }
-                                    if !is_valid_b {
-                                        cmp = Ordering::Greater;
-                                    }
-                                    if options.descending
-                                        | ((!is_valid_a | !is_valid_b) & options.nulls_last)
-                                    {
+                                    if options.descending {
                                         cmp = cmp.reverse();
                                     }
                                     cmp

--- a/py-polars/tests/unit/operations/test_over.py
+++ b/py-polars/tests/unit/operations/test_over.py
@@ -183,6 +183,15 @@ def test_nulls_last_over_24989() -> None:
     assert_frame_equal(out, expected)
 
 
+def test_over_order_by_descending_nulls_agg_context() -> None:
+    df = pl.DataFrame({"g": ["a", "a", "a"], "v": [1, 2, 3], "d": [1, None, 2]})
+    expr = pl.col("v").cum_sum().over("g", order_by="d", descending=True)
+
+    result = df.group_by("g", maintain_order=True).agg(expr.alias("v"))
+
+    assert result["v"][0].to_list() == [6, 2, 5]
+
+
 def test_over_duplicate_partition_by_26921() -> None:
     df = pl.DataFrame({"x": [1, 2, 3]})
     with pytest.raises(pl.exceptions.DuplicateError):

